client/sites/components/sites-dataviews/index.tsx

```diff
@@ -173,12 +189,60 @@ const DotcomSitesDataViews = ( {
 		viewType: dataViewsState.type,
 	} );
 
+	const getTracksEventDataForFilter = ( fieldId: FieldId, value: any ): TracksEventData | null => {
+		if ( value === undefined ) {
+			return null;
+		}
+
+		switch ( fieldId ) {
+			case 'status': {
+				const statusGroup = siteStatusGroups.find( ( group ) => group.value === value );
+				return [ 'calypso_sites_dashboard_filter_status', statusGroup?.slug ?? '' ];
+			}
+
+			case 'icon':
+			case 'last-interacted':
+			case 'last-publish':
+			case 'plan':
+			case 'site-title':
+			case 'stats':
+				return null;
+
+			default: {
+				// Ensures that the switch statement is exhaustive
+				const exhaustiveCheck: never = fieldId;
+				return exhaustiveCheck;
+			}
+		}
+	};
+
+	const previousViewFilters = useRef< Record< string, string > >( {} );
+
+	const submitTracksEventForFilters = ( filters: Filter[] ) => {
+		const filterTracksEvents = filters
+			.map( ( filter ) => getTracksEventDataForFilter( filter.field as FieldId, filter.value ) )
+			.filter( ( filter ): filter is TracksEventData => filter !== null );
+
+		filterTracksEvents.forEach( ( [ name, value ] ) => {
+			if ( previousViewFilters.current[ name ] !== value ) {
+				recordTracksEvent( name, {
+					value: value,
+				} );
+			}
+		} );
```

Comment:
This is a personal preference/opinion, but I believe we can improve the code's readability by adopting a more functional approach, like this:

```
filterTracksEvents
	.filter(([name, value]) => previousViewFilters.current[name] !== value)
	.forEach(([name, value]) => {
		recordTracksEvent(name, { value });
	});
```